src/crystal/system/win32/file.cr

```diff
@@ -26,8 +26,9 @@ module Crystal::System::File
     fd
   end
 
-  def self.mktemp(prefix : String, suffix : String?, dir : String) : {LibC::Int, String}
-    path = "#{tempdir}\\#{prefix}.#{::Random::Secure.hex}#{suffix}"
+  def self.mktemp(prefix : String?, suffix : String?, dir : String) : {LibC::Int, String}
+    dir = dir + ::File::SEPARATOR
+    path = "#{dir}#{prefix}.#{::Random::Secure.hex}#{suffix}"
```

Comment:
This avoids an extra allocation:
```suggestion
    path = "#{dir}#{::File::SEPARATOR}#{prefix}.#{::Random::Secure.hex}#{suffix}"
```
Or use `::File.join` I think we can depend on that from here.

Reply:
This is not by accident, it's just copying the implementation from unix. I'd prefer to not refactor that here (pretend that I'm "not changing the code"); maybe someone can do both at the same time.

```
$ grep -rF1 'dir = dir + ::File::SEPARATOR'
src/crystal/system/unix/file.cr-  def self.mktemp(prefix, suffix, dir) : {LibC::Int, String}
src/crystal/system/unix/file.cr:    dir = dir + ::File::SEPARATOR
src/crystal/system/unix/file.cr-    path = "#{dir}#{prefix}.XXXXXX#{suffix}"
--
src/crystal/system/win32/file.cr-  def self.mktemp(prefix : String?, suffix : String?, dir : String) : {LibC::Int, String}
src/crystal/system/win32/file.cr:    dir = dir + ::File::SEPARATOR
src/crystal/system/win32/file.cr-    path = "#{dir}#{prefix}.#{::Random::Secure.hex}#{suffix}"
```

Reply:
Ah whatever, applied it. Thanks. Leaving it wouldn't make as much sense as I make it seem.